packages/connect-form/src/components/advanced-options-tabs/advanced-tab/url-options-table.tsx

```diff
@@ -0,0 +1,251 @@
+import React, { ChangeEvent, useEffect } from 'react';
+import { css } from '@emotion/css';
+import {
+  spacing,
+  Table,
+  TableHeader,
+  Row,
+  Cell,
+  Button,
+  IconButton,
+  Icon,
+  TextInput,
+  Select,
+  Option,
+  OptionGroup,
+  Banner,
+} from '@mongodb-js/compass-components';
+import ConnectionStringUrl from 'mongodb-connection-string-url';
+
+import { editableUrlOptions, UrlOption } from '../../../utils/url-options';
+import { UpdateConnectionFormField } from '../../../hooks/use-connect-form';
+
+const tableHeaderStyles = css({
+  width: '50%',
+});
+
+const optionValueCellStyles = css({
+  display: 'flex',
+  flexDirection: 'row',
+  justifyContent: 'space-between',
+  alignItems: 'center',
+  width: '100%',
+});
+
+const addUrlOptionsButtonStyles = css({
+  textAlign: 'center',
+  marginTop: spacing[3],
+  marginBottom: spacing[2],
+});
+
+function getUrlOptions(connectionStringUrl: ConnectionStringUrl): UrlOption[] {
+  let opts: string[] = [];
+  const urlOptions: UrlOption[] = [];
+
+  editableUrlOptions.forEach(({ values }) => (opts = opts.concat(values)));
+  connectionStringUrl.searchParams.forEach((value, name) => {
+    if (opts.includes(name)) {
+      urlOptions.push({
+        value: value,
+        name: name as UrlOption['name'],
+      });
+    }
+  });
+  return urlOptions;
+}
+
+function UrlOptionsTable({
+  updateConnectionFormField,
+  connectionStringUrl,
+}: {
+  updateConnectionFormField: UpdateConnectionFormField;
+  connectionStringUrl: ConnectionStringUrl;
+}): React.ReactElement {
+  const [options, setOptions] = React.useState<Partial<UrlOption>[]>([]);
+  const [errorMessge, setErrorMessge] = React.useState('');
+
+  useEffect(() => {
+    const options: Partial<UrlOption>[] = getUrlOptions(connectionStringUrl);
+    if (!options.length) {
+      options.push({ name: undefined, value: undefined });
+    }
+    setOptions(options);
+  }, [connectionStringUrl]);
+
+  const addUrlOption = () => {
```

Comment:
We can probably use `useCallback` here and also int he other two functions below to avoid extra re-renders.
Here's an example: https://github.com/mongodb-js/compass/blob/main/packages/connect-form/src/components/advanced-options-tabs/general/direct-connection-input.tsx#L20

Reply:
using `useCallback` here has a dependency on `options` which is a state parameter and any interaction in this component (to change value, delete or add new), changes the state of `options`